core/src/main/java/io/opencensus/stats/NoopStats.java

```diff
@@ -88,10 +99,28 @@ public StatsRecorder getStatsRecorder() {
     static final StatsRecorder INSTANCE = new NoopStatsRecorder();
 
     @Override
-    public void record(TagContext tags, MeasureMap measureValues) {
+    public StatsBuilder newStatsBuilder(TagContext tags) {
       checkNotNull(tags, "tags");
-      checkNotNull(measureValues, "measureValues");
+      return getNoopStatsBuilder();
+    }
+  }
+
+  @Immutable
+  private static final class NoopStatsBuilder extends StatsBuilder {
+    static final StatsBuilder INSTANCE = new NoopStatsBuilder();
+
+    @Override
+    public StatsBuilder put(MeasureDouble measure, double value) {
+      return this;
+    }
+
+    @Override
+    public StatsBuilder put(MeasureLong measure, long value) {
+      return this;
     }
+
+    @Override
+    public void record() {}
```

Comment:
because Java I may call this "buildAndRecord". You may keep a build as well and the record function:
1. If we want to count active/finished requests we will always record a long metric with value 1 so I can allocate that object once.
2. I think there are other examples like this when you record the same metrics every time.

Reply:
I didn't consider that users might need to record the same values twice.  Do you think that would be likely for a map with size greater than 1?  I think we could handle the 1 measure case by adding a convenience method like `record(MeasureDouble, double)`.

If we also added a `build` method, it might be better to separate it from the `record` method, something like

```java
MeasureMap measures =
     statsRecorder.measureMapBuilder().
            .put(RPC_CLIENT_ROUNDTRIP_LATENCY, 100)
            .put(RPC_CLIENT_REQUEST_BYTES, 2000)
            .build();
statsRecorder.record(measures);
```

/cc @dinooliva